File: in_Python/0500 Keyboard Row.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        res = []
        dict = {}
        l1 = ['q','w','e','r','t','y','u','i','o','p']
        l2 = ['a','s','d','f','g','h','j','k','l']
        l3 = ['z','x','c','v','b','n','m']
        for x in l1:
            dict[x] = 1
        for x in l2:
            dict[x] = 2
        for x in l3:
            dict[x] = 3
        for w in words:
            flag = False
            temp = []
            for i in list(w):
                if dict[i.lower()] not in temp:
                    temp.append(dict[i.lower()])
                if len(temp) == 2:
                    flag = True
                    break
            if not flag:
                res.append(w)
        return res

File: in_Java/0500 Keyboard Row.java
class Solution {
    public String[] findWords(String[] words) {
        char[] r1 = new char[]{'q','w','e','r','t','y','u','i','o','p'};
        char[] r2 = new char[]{'a','s','d','f','g','h','j','k','l'};
        char[] r3 = new char[]{'z','x','c','v','b','n','m'};
        HashMap<Character,Integer> hm = new HashMap<>();
        for(char c : r1) hm.put(c,1);
        for(char c : r2) hm.put(c,2);
        for(char c : r3) hm.put(c,3);
        List<String> ar = new ArrayList<>();
        boolean f;
        for(String ss : words){
            String s = ss.toLowerCase();
            f = true;
            int rowid = hm.get(s.charAt(0));
            for(int i = 1; i < s.length(); i++){
                if(hm.get(s.charAt(i)) != rowid){
                    f = false;
                    break;
                }
            }
            if(f) ar.add(ss);
        }
        String[] res = new String[ar.size()];
        for(int i = 0; i < res.length; i++){
            res[i] = ar.get(i);
        }
        return res;
    }
}